### ISSUE TITLE
Filter policy support for SNS events

### ISSUE THREAD

**Issue by joshm91** [awslabs/serverless-application-model#261]
Hi guys,

It'd be great to be able to support the new(-ish) [Filter Policy](https://docs.aws.amazon.com/sns/latest/dg/message-filtering.html) feature in SNS when specifying an SNS event on a Lambda function.  

Something like this would be perfect:

```yaml
  myLambda:
    Type: AWS::Serverless::Function
    Properties:
      Handler: lambda_handlers.handler
      ...
      Events:
        SNS:
          Type: SNS
          Properties:
            Topic: !GetAtt someSnsTopic.Arn
            FilterPolicy:
              AttributeName1:
                - value1
                - value2
              AttributeName2:
                - value3
```

**Comment by jfuss** [comment]
@joshm91 Filter Policy is not available for CloudFormation just yet and therefore not able to be apart of SAM. Marking this as a feature request.

**Comment by Nr18** [comment]
Just yet 🤔so it’s coming 🤪

**Comment by joshm91** [comment]
Faur enough, thanks!

**Comment by oharaandrew314** [comment]
Yes, this would be great.  I have several lambdas listening to one SNS topic, and I would vastly prefer being able to filter outside of the lambda invocation, which skews my invocation counts.

**Comment by kevohagan** [comment]
@jfuss is this available in CloudFormation now? this would be great in sam! thx

**Comment by vrivellino** [comment]
It looks like it's available now: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/ReleaseHistory.html

**Comment by bottemav** [comment]
@joshm91 In the CloudFormation template one can also filter on a prefix or a numeric value matching. Do you need this too or does whitelisting (exact string matching) satisfy your use case?

**Comment by brettstack** [comment]
Happy to take PRs on this.

**Comment by brettstack** [comment]
Thanks!

**Comment by otaviofff** [comment]
Hey guys. How is it coming together? Thanks a lot

**Comment by brettstack** [comment]
We've just merged this into develop branch and will go out in our next release. Thanks @bottemav 🙌

**Comment by otaviofff** [comment]
Fantastic! Thanks for adding the suppport for SNS message filtering.

**Comment by brettstack** [comment]
Closing this since it's been merged into develop and will be released in the next release.

**Comment by davidtwamley** [comment]
@brettstack what version will this show up in? I just upgraded with homebrew to 0.9.0 and I'm still getting a "property FilterPolicy not defined for resource of type SNS" error. Thanks

**Comment by jfuss** [comment]
FilterPolicy is not released so SAM CLI doesn't know about this. 

When SAM Translator (this repo) updates/cuts a new version with the FilterPolicy in it, SAM CLI will need to cut a release and do a version bump to that version. Until then, this is not available.

**Comment by studds** [comment]
When will the next version be released?

**Comment by davidtwamley** [comment]
I'm eagerly waiting on this one. Several sessions at re:Invent (3 months ago now) sold me on using this idea for enabling loosely coupled microservices. I'm using an ugly hack for now to mimic this feature where I do my own filtering in the listeners. 20 lambdas answer and 19 hang up immediately. Not scalable and it probably interferes with the SNS retry logic. If possible, please pressure management to deliver on last year's re:Invent goodness soon. Thanks!

**Comment by keetonian** [comment]
Released!

**Comment by davidtwamley** [comment]
Works great! Thank you all very much.
